[Concept Entry] Rust loops #3374 (#3856)
Codecademy/docs: content/rust/concepts/loops/loops.md

---
Title: 'Loops'
Description: 'A loop is a control structure that allows you to repeatedly execute a block of code.'
Subjects:

>   - 'Computer Science'

Reviewer: ```suggestion
  - 'Computer Science'
  - 'Code Foundations'
```

Tags:
  - 'Loops'
CatalogContent:
  - 'Learn-rust'
  - 'paths/computer-science'
---

> A `loop` is a control structure that allows you to execute a block of code repeatedly. Loops are used to perform tasks that need to be done multiple times, such as iterating through a list of items, performing calculations, or waiting for a specific condition to be met.There typically three main types `for`, `while` and `loop`.

Reviewer: ```suggestion
A **`loop`** is a control structure that enables the repetitive execution of a block of code. `Loops` are employed for tasks that require multiple iterations, such as iterating through a list of items, performing calculations, or waiting for a specific condition to be met. There are typically three main types: `for`, `while`, and `loop`.
```
we don't prefer to use pronouns. Refrain form using them.

> 

Reviewer: ```suggestion
## Syntax

```
We always add syntax heading then we add the example heading. We can make the different kinds of loops as sub headings.

> ## for loops

Reviewer: ```suggestion
### For loop
```

> The `for` loop is used to iterate over collections or ranges.It automatically iterates through elements and executes a block of code for each item in the collection or range.It's commonly used for iterating over arrays, vectors, and other iterable data structures.

Reviewer: ```suggestion
The `for loop` is used to iterate over collections or ranges. It automatically iterates through elements and executes a code block for each item in the pool or scope. It's commonly used for iterating over `arrays`, `vectors`, and other iterable data structures.
```

```rust
let numbers = ["A", 2,20, 4, 5];
for num in numbers.iter() {
    // Code to be executed for each element in the array
    println!("Number: {}", num);
}
```

> ## while loops

Reviewer: ```suggestion
### While loop
```

> The `while` loop repeatedly executes a block of code as long as a specified condition remains true.It checks the condition before entering the loop, and if the condition is initially false, the loop won't run.While loops are useful when you don't know in advance how many iterations are needed.

Reviewer: ```suggestion
The `while loop` repeatedly executes a block of code as long as a specified condition remains true. It checks the condition before entering the loop, and if the condition is initially `false`, the loop won't run. `While loops` are useful when you don't know in advance how many iterations are needed.
```

```rust
let mut counter = 0;
while counter < 5 {
    // Code to be executed while the condition is true
    counter += 1;
}
```

> ## loop loops

Reviewer: ```suggestion
### Loop 
```

> The `loop` keyword creates an infinite loop that continues executing the code block indefinitely until it's explicitly interrupted using the break statement.It's useful when you need to perform a task until a certain condition is met within the loop.

Reviewer: ```suggestion
The `loop` keyword creates an infinite loop that continues executing the code block indefinitely until it's explicitly interrupted using the break statement. It's useful when you need to perform a task until a certain condition is met within the loop.
```

```rust
fn main() {
    loop {
        println!("Enter 'exit' to quit: ");
        let mut input = String::new();
        std::io::stdin().read_line(&mut input).expect("Failed to read line");

        if input.trim() == "exit" {
            break;
        }
    }
}
```

## Example

> This example demonstrates a `while` loop that iterates from `5` to `10` as the counter variable starts from `5`.The second example shows a `for` loop that iterates over an array that containe numbers `1` to `5` .

Reviewer: ```suggestion
This example demonstrates a `while loop` that iterates from `5` to `10` as the counter variable starts from `5`. The second example shows a `for` loop that iterates over an `array` that contains numbers `1` to `5`.
```

---
Title: 'Loops'
Description: 'A loop is a control structure that allows you to repeatedly execute a block of code.'
Subjects:
  - 'Computer Science'
  - 'Code Foundations'
Tags:
  - 'Loops'
CatalogContent:

>   - 'Learn-rust'

Reviewer 2: ```suggestion
  - 'rust-for-programmers'
```

## Syntax

### For loop

The `for loop` is used to iterate over collections or ranges. It automatically iterates through elements and executes a code block for each item in the pool or scope. It's commonly used for iterating over `arrays`, `vectors`, and other iterable data structures.

```pseudo
let numbers = ["A", 2,20, 4, 5];
for num in numbers.iter() {
    // Code to be executed for each element in the array
    println!("Number: {}", num);
}
```

### While loops

The `while` loop repeatedly executes a block of code as long as a specified condition remains true. It checks the condition before entering the loop, and if the condition is initially false, the loop won't run. While loops are useful when the number of iterations is not known in advance.

```pseudo
let mut counter = 0;
while counter < 5 {
    // Code to be executed while the condition is true
    counter += 1;
}
```

> ### Loop

Author: Hello @caupolicandiaz If the sub heading on line 42 changes to infinite loop should we also change the intro at line 14 ? . Suggested change :A **loop** is a control structure that enables the repetitive execution of a block of code. Loops are employed for tasks that require multiple iterations, such as iterating through a list of items, performing calculations, or waiting for a specific condition to be met. There are typically three main types: `for`, `while` and  infinite `loop`. Thank you

The `loop` keyword creates an infinite loop that continues executing the code block indefinitely until it's explicitly interrupted using the break statement. It's useful when a task must be performed until a certain condition is met within the loop.

```pseudo
fn main() {
    loop {
        println!("Enter 'exit' to quit: ");
        let mut input = String::new();
        std::io::stdin().read_line(&mut input).expect("Failed to read line");

        if input.trim() == "exit" {
            break;
        }
    }
}
```

## Example

> This example demonstrates a `while loop` that iterates from `5` to `10` as the counter variable starts from `5`. The second example shows a `for` loop that iterates over an `array` that contains numbers `1` to `5`.

Reviewer 2: ```suggestion
This example demonstrates a `while loop` that iterates through each integer from `5` to `10`. The second example uses a `for` loop to iterate over an array that contains the integers in the range `1` to `5`.
```
